KenticoCloud.Delivery/DeliveryClient.cs

```diff
@@ -590,5 +589,30 @@ private async Task<JObject> GetResponseContent(HttpResponseMessage httpResponseM
 
             throw new DeliveryException(httpResponseMessage, faultContent);
         }
+
+        internal IEnumerable<IQueryParameter> ExtractParameters<T>(IEnumerable<IQueryParameter> parameters = null)
+        {
+            var enhancedParameters = parameters != null
+                ? new List<IQueryParameter>(parameters)
+                : new List<IQueryParameter>();
+
+            var codename = _codeFirstModelProvider.TypeProvider.GetCodename(typeof(T));
+
+            if (!IsAlreadyInParameters(parameters) && codename != null)
```

Comment:
Performance hint: Switch the check, if codename is null there is no need to check the params.